src/Reflection/ReflectionClass.php

```diff
@@ -242,7 +242,15 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $name = 'class@anonymous';
+        if ($this->getFileName() !== null) {
+            $name .= ' ' . $this->getFileName();
```

Comment:
Note: it may be that this is acceptable, because in real terms the class name is irrelevant anyway :)